tests/plugins/test_pin_operator_digests.py

```diff
@@ -1352,6 +1354,342 @@ def test_not_allowed_attributes_for_modifications(self, tmpdir, docker_tasker, m
+                                    }
+                                }],
+                                'fake_object': {
+                                    'spec': {
+                                        'fake_list': ['fake_item']
+                                    }
+                                }
+                            }
+                        },
+                        'relatedImages': [{'name': 'foo-1', 'image': 'myimage:v1.2.2'}]
+                    }
+                },
+                id='append_only_deep'
+            )
+        ]
+    )
+    @responses.activate
+    def test_other_metadata_replacements_pass(self, tmpdir, docker_tasker,
+                                              mods,
+                                              allowed_attrs,
+                                              expected_final_csv):
+        csv_yaml = 'csv.yaml'
+        pull_specs = ['myimage:v1.2.2']
+        test_url = 'https://example.com/modifications.json'
+        modification_data = {
+            'pullspec_replacements': PULLSPEC_REPLACEMENTS
+        }
+        modification_data.update(mods)
+
+        responses.add(responses.GET, test_url, json=modification_data)
+
+        manifests_dir = tmpdir.join(OPERATOR_MANIFESTS_DIR).mkdir()
+        mock_operator_csv(manifests_dir, csv_yaml, pull_specs)
+
+        user_config = get_user_config(OPERATOR_MANIFESTS_DIR)
+        site_config = get_site_config(
+            operator_csv_modifications_allowed_attributes=allowed_attrs,
+        )
+
+        runner = mock_env(
+            docker_tasker, tmpdir, orchestrator=False,
+            user_config=user_config,
+            site_config=site_config,
+            operator_csv_modifications_url=test_url
+        )
+
+        pytest_tmp_dir = runner.workflow.source.source_path
+
+        runner.run()
+
+        with open(os.path.join(pytest_tmp_dir, OPERATOR_MANIFESTS_DIR, csv_yaml), 'r') as csv:
+            final_csv = yaml.load(csv.read())
+
+        assert final_csv == expected_final_csv
+
+    @pytest.mark.parametrize(
+        'mods,allowed_attrs,exc_msg',
+        [
+            pytest.param(
+                {
+                    'append': {'spec': {'skips': '1.2.3'}}
+                },
+                [
+                    ['spec', 'skips']
+                ],
+                (
+                    "{'skips': '1.2.3'} is not of type 'array'"
+                ),
+                id='append_not_list'
+            ),
+            pytest.param(
+                {
+                    'update': [{'foo': {'bar': ['1.2.3']}}]
+                },
+                [
+                    ['foo']
+                ],
+                (
+                    "[{'foo': {'bar': ['1.2.3']}}] is not of type 'object'"
+                ),
+                id='update_not_dict'
+            ),
+        ]
+    )
+    @responses.activate
+    def test_other_metadata_replacements_fail(self, tmpdir, docker_tasker,
```

Comment:
I wouldn't test again failing on validation, just that case when you are trying to extend string

Reply:
Agreed